### PR TITLE
BCR presubmit: Support testing test module

### PR DESCRIPTION
**Pull request by meteorcloudy**
Implementing https://github.com/bazelbuild/continuous-integration/issues/1302

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Test run for https://github.com/meteorcloudy/bazel-central-registry/pull/7 is here: https://buildkite.com/bazel-testing/bcr-presubmit/builds/67

File: buildkite/bazelci.py [hunk 1]
```diff
@@ -873,6 +875,54 @@ def downstream_projects_root(platform):
...
+
+
+def get_combinations(matrix, attributes):
+    """Given a matrix and the wanted attributes, return all possible combinations.
+
+    eg.
+    With matrix = {'a': [1, 2], 'b': [1], 'c': [2]},
+    if attributes = ['a', 'b'], then returns [[('a', 1), ('b', 1)], [('a', 2), ('b', 1)]]
+    if attributes = ['b', 'c'], then returns [[('b', 1), ('c', 1)]]
+    if attributes = ['c'], then returns [[('c', 1)]]
```
Comment (Wyverald) [inline buildkite/bazelci.py:904]: "then returns [[('c', **2**)]]"?

Comment (meteorcloudy) [reply buildkite/bazelci.py:904]: Nice catch, but I'll fix the matrix instead.

Comment (meteorcloudy) [reply buildkite/bazelci.py]: Also rebase with master branch after merging https://github.com/bazelbuild/continuous-integration/pull/1300

File: buildkite/bazel-central-registry/bcr_presubmit.py [hunk 2]
```diff
@@ -140,13 +170,85 @@ def create_test_repo(module_name, module_version, task):
...
+    root = get_root_dir(module_name, module_version, task, is_test_module = True)
+    source = load_source_json(module_name, module_version)
+
+    # Download and unpack the source archive to ./output
+    archive_url = source["url"]
+    archive_file = root.joinpath(archive_url.split("/")[-1])
+    output_dir = root.joinpath("output")
+    bazelci.eprint("* Download and unpack %s\n" % archive_url)
+    download(archive_url, archive_file)
+    shutil.unpack_archive(str(archive_file), output_dir)
```
Comment (seh) [inline buildkite/bazel-central-registry/bcr_presubmit.py:209]: At present, we lose the permissions on the extracted files, per python/cpython#59999.

**Pull request merged**